add a new blog for etcd learner mode usage in kubeadm join (#43095)
kubernetes/website: content/en/blog/_posts/2023-09-25-kubeadm-use-etcd-learner-mode.md

## How nodes join a cluster that's using the new mode

### Create a Kubernetes cluster backed by etcd in learner mode {#create-K8s-cluster-etcd-learner-mode}

For a general explanation about creating highly available clusters with kubeadm, you can refer to
[Creating Highly Available Clusters with kubeadm](/docs/setup/production-environment/tools/kubeadm/high-availability/).

To create a Kubernetes cluster, backed by etcd in learner mode, using kubeadm, follow these steps:

```shell
# kubeadm init --feature-gates=EtcdLearnerMode=true ...
kubeadm init --config=kubeadm-config.yaml
```

The kubeadm configuration file is like below:

```yaml
apiVersion: kubeadm.k8s.io/v1beta3
kind: ClusterConfiguration
featureGates:
  EtcdLearnerMode: true
```

The kubeadm tool deploys a single-node Kubernetes cluster with etcd set to use learner mode.

### Join nodes to the Kubernetes cluster

Before joining a control-plane node to the new Kubernetes cluster, ensure that the existing control plane nodes
and all etcd members are healthy.

Check the cluster health with `etcdctl`. If `etcdctl` isn't available, you can run this tool inside a container image.

> You would do that directly with your container runtime using a tool such as crictl run and not through Kubernetes

Reviewer: ```suggestion
You would do that directly with your container runtime using a tool such as `crictl run` and not through Kubernetes
```

Here is an example on a client command that uses secure communication to check the cluster health of the etcd cluster:

```shell
ETCDCTL_API=3 etcdctl --endpoints 127.0.0.1:2379 \
  --cert=/etc/kubernetes/pki/etcd/server.crt \
  --key=/etc/kubernetes/pki/etcd/server.key \
  --cacert=/etc/kubernetes/pki/etcd/ca.crt \
   member list
...
dc543c4d307fadb9, started, node1, https://10.6.177.40:2380, https://10.6.177.40:2379, false
```

> To check the Kubernetes level control-plane healthy, run `kubectl get node -l node-role.kubernetes.io/control-plane=`

Reviewer: ```suggestion
To check if the Kubernetes control plane is healthy, run `kubectl get node -l node-role.kubernetes.io/control-plane=`
```

> To check the Kubernetes level control-plane healthy, run `kubectl get node  -l node-role.kubernetes.io/control-plane=`

Reviewer 2: ```suggestion
To check the Kubernetes level control-plane healthy, run `kubectl get node -l node-role.kubernetes.io/control-plane=`
```

> and check node are ready.

Reviewer 2: ```suggestion
and check the nodes are ready.
```

Author: Thanks for the review. Updated.

> and check the nodes are ready.

Reviewer: ```suggestion
and check if the nodes are ready.
```

Note: It is recommended to have an odd number of members in a etcd cluster.

Before joining a worker node to the new Kubernetes cluster, ensure that the control plane nodes are healthy.

## What's next

> - The feature gate `EtcdLearnerMode` is alpha in v1.27 and will we expect it to graduate beta in the next

Reviewer: ```suggestion
- The feature gate `EtcdLearnerMode` is alpha in v1.27 and we expect it to graduate to beta in the next
```
